programs/rewards/src/state/mining.rs

```diff
@@ -199,25 +209,29 @@ impl Mining {
     }
 
     pub fn restrict_tokenflow(&mut self) -> ProgramResult {
-        if self.data[CLAIMING_RESTRICTION_BYTE] == 1 {
+        if self.is_tokenflow_restricted() {
             Err(MplxRewardsError::MiningAlreadyRestricted.into())
         } else {
-            self.data[CLAIMING_RESTRICTION_BYTE] = 1;
+            self.data[PENALTIES_BYTE] |= 1;
             Ok(())
         }
     }
 
     pub fn allow_tokenflow(&mut self) -> ProgramResult {
-        if self.data[CLAIMING_RESTRICTION_BYTE] == 0 {
-            Err(MplxRewardsError::MiningNotRestricted.into())
+        if !self.is_tokenflow_restricted() {
+            Err(MplxRewardsError::MiningAlreadyRestricted.into())
         } else {
-            self.data[CLAIMING_RESTRICTION_BYTE] = 0;
+            self.data[PENALTIES_BYTE] &= 0;
```

Comment:
This will drop all the bits and set those to zero.
```suggestion
            self.data[PENALTIES_BYTE] &= !(1 << 0);
```
This last part may be refactored to 
```
const IS_TOKENFLOW_RESTRICTED_MASK:u8 = 1<<0;
```
to be used in checks